.github/workflows/tests.yml

```diff
@@ -0,0 +1,289 @@
+          postgres:${{ matrix.postgres-version }}
+    - uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python-version }}
+    - run: pip install tox
+    - name: Await PostgreSQL
+      if: ${{ matrix.postgres-version }}
+      timeout-minutes: 2
+      run: until pg_isready -h localhost; do sleep 1; done
+    - run: tox -e py,combine
+      env:
+        TRIAL_FLAGS: "--jobs=2"
+        SYNAPSE_POSTGRES: ${{ matrix.database == 'postgres' || '' }}
+        SYNAPSE_POSTGRES_HOST: localhost
+        SYNAPSE_POSTGRES_USER: postgres
+        SYNAPSE_POSTGRES_PASSWORD: postgres
+
+
+  trial-olddeps:
+    if: ${{ !failure() }}  # Allow previous steps to be skipped, but not fail
+    needs: linting-done
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: Test with old deps
+      uses: docker://ubuntu:xenial  # For old python and sqlite
+      with:
+        workdir: /github/workspace
+        entrypoint: .buildkite/scripts/test_old_deps.sh
+      env:
+        TRIAL_FLAGS: "--jobs=2"
+
+
+  trial-pypy:
+    # Very slow; only run if the branch name includes 'pypy'
+    if: ${{ contains(github.ref, 'pypy') && !failure() }}
+    needs: linting-done
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ['pypy-3.6']
+
+    steps:
+    - uses: actions/checkout@v2
+    - run: sudo apt-get -qq install xmlsec1 libxml2-dev libxslt-dev
+    - uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python-version }}
+    - run: pip install tox
+    - run: tox -e py,combine
+      env:
+        TRIAL_FLAGS: "--jobs=2"
+
+
+  sytest:
+    if: ${{ !failure() }}
+    needs: linting-done
+    runs-on: ubuntu-latest
+    container:
+      image: matrixdotorg/sytest-synapse:${{ matrix.sytest-tag }}
+      volumes:
+        - ${{ github.workspace }}:/src
+      env:
+        BUILDKITE_BRANCH: ${{ github.head_ref }}
+        POSTGRES: ${{ matrix.postgres && 1}}
+        MULTI_POSTGRES: ${{ (matrix.postgres == 'multi-postgres') && 1}}
+        WORKERS: ${{ matrix.workers && 1 }}
+        REDIS: ${{ matrix.redis && 1 }}
+        BLACKLIST: ${{ matrix.workers && 'synapse-blacklist-with-workers' }}
+
+    strategy:
+      matrix:
+        include:
+          - sytest-tag: py35
+
+          - sytest-tag: py35
+            postgres: postgres
+
+          - sytest-tag: py38
+            postgres: postgres
+
+          - sytest-tag: py35
+            postgres: multi-postgres
+            workers: workers
+
+          - sytest-tag: py37
+            postgres: multi-postgres
+            workers: workers
+
+          - sytest-tag: py37
+            postgres: postgres
+            workers: workers
+            redis: redis
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Prepare test blacklist
+      run: cat sytest-blacklist .buildkite/worker-blacklist > synapse-blacklist-with-workers
+    - name: Run SyTest
+      continue-on-error: true  # FIXME
```

Comment:
why's this a `FIXME`?

Reply:
If one SyTest fails we should fail the whole matrix... but currently SyTests are a bit flakey :(